src/transformers/pipelines/token_classification.py

```diff
@@ -234,56 +234,68 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
             return answers[0]
         return answers
 
-    def set_subwords_label(self, entities: List[dict], strategy: str) -> dict:
-        def sub_words_label(sub_words: List[dict]) -> dict:
+    def set_subwords_label(self, entities: List[dict]) -> List[dict]:
+        strategy = self.subword_label_re_alignment
+
+        def set_labels(sub_words: List[dict]) -> dict:
             score = np.stack([sub["score"] for sub in sub_words])
-            if strategy == "default":
+            if strategy == "default" or strategy is True:
                 label_idx = score[0].argmax()
                 label = self.model.config.id2label[label_idx]
                 sub_words[0]["entity"] = label
+                sub_words[0]["score"] = score[0][label_idx]
                 for sub in sub_words[1:]:
                     sub["entity"] = "O"
-                return sub_words
+                    sub["score"] = 0.0  # what score should we assign here?
+            else:
+                if strategy == "first":
+                    # get label of first sub-word
+                    max_label_idx = score[0].argmax()
+                    label = self.model.config.id2label[max_label_idx]
+                elif strategy == "max":
+                    max_label_idx = np.unravel_index(np.argmax(score, axis=None), score.shape)[1]
+                    label = self.model.config.id2label[max_label_idx]
+                elif strategy == "average":
+                    max_label_idx = np.mean(score, axis=0).argmax()
+                    label = self.model.config.id2label[max_label_idx]
+                else:
+                    raise ValueError(f"Invalid value {strategy} for option `subword_label_re_alignment`")
 
-            if strategy == "first":
-                # get label of first sub-word
-                label_idx = score[0].argmax()
-                label = self.model.config.id2label[label_idx]
-            elif strategy == "max":
-                max_label_idx = np.unravel_index(np.argmax(score, axis=None), score.shape)[1]
-                label = self.model.config.id2label[max_label_idx]
-            elif strategy == "average":
-                max_label_idx = np.mean(score, axis=0).argmax()
-                label = self.model.config.id2label[max_label_idx]
+                for idx, sub in enumerate(sub_words):
+                    sub["entity"] = label
+                    sub["score"] = score[idx][max_label_idx].item()
 
-            for idx, sub in enumerate(sub_words):
-                sub["entity"] = label
-                sub["score"] = score[idx][max_label_idx].item()
+            if self.ignore_subwords:
```

Comment:
I think we don't need to check `self.ignore_subwords` here, we just want to set labels for subwords in `set_labels` function. later according to `self.grouped_entities` we'll decide to merge subwords or not. take a look at [this](https://github.com/huggingface/transformers/issues/10263#issuecomment-782187601) as joshdevins said, I also don't get what is the purpose of `self.ignore_subwords` in the first place. what are your reasons to check `self.ignore_subwords` here ?

Reply:
The problem is that `grouped_entities` merges _all_ the consecutive tokens with the same label, not just subwords. In my use case, I need to merge subwords without aggregating multiple words in the same entity.
Considering the example from that thread:
```Max Mustermann → Max Must ##erman ##n```
with `grouped_entities=True and ignore_subwords=True` we get a single entity `"Max Mustermann"`
with `ignore_subwords=True and label_re_alignment=True` we can keep the two words as separate entities `["Max","Mustermann"]`.
This is useful when you need to relabel with additional attributes (in this case it could be `FIRST_NAME`, `LAST_NAME`, for example), or more generally perform subsequent tasks at the word level.